src/main/java/org/sonarlint/intellij/ui/SonarLintHotspotsListPanel.java

```diff
@@ -19,99 +19,201 @@
 import com.intellij.openapi.project.Project;
 import com.intellij.openapi.roots.ProjectRootManager;
-import com.intellij.ui.CollectionListModel;
+import com.intellij.openapi.ui.SimpleToolWindowPanel;
+import com.intellij.openapi.ui.VerticalFlowLayout;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.tools.SimpleActionGroup;
 import com.intellij.ui.ScrollPaneFactory;
-import com.intellij.ui.components.JBList;
+import com.intellij.ui.components.JBTabbedPane;
 import com.intellij.ui.components.labels.ActionLink;
 import com.intellij.ui.components.panels.HorizontalLayout;
-import com.intellij.util.EditSourceOnDoubleClickHandler;
-import com.intellij.util.EditSourceOnEnterKeyHandler;
+import com.intellij.ui.treeStructure.Tree;
+import com.intellij.util.ui.tree.TreeUtil;
 import java.awt.BorderLayout;
+import java.awt.event.KeyAdapter;
+import java.awt.event.KeyEvent;
 import java.util.Collection;
+import java.util.Map;
 import javax.annotation.Nullable;
+import javax.swing.Box;
 import javax.swing.JComponent;
 import javax.swing.JLabel;
 import javax.swing.JPanel;
+import javax.swing.event.TreeSelectionEvent;
+import javax.swing.tree.TreeSelectionModel;
 import org.sonarlint.intellij.actions.SonarConfigureProject;
-import org.sonarlint.intellij.config.global.ServerConnection;
-import org.sonarlint.intellij.core.ProjectBindingManager;
+import org.sonarlint.intellij.common.util.SonarLintUtils;
 import org.sonarlint.intellij.editor.EditorDecorator;
-import org.sonarlint.intellij.exception.InvalidBindingException;
-import org.sonarlint.intellij.finding.hotspot.FoundSecurityHotspots;
+import org.sonarlint.intellij.finding.LiveFinding;
 import org.sonarlint.intellij.finding.hotspot.InvalidBinding;
 import org.sonarlint.intellij.finding.hotspot.LiveSecurityHotspot;
 import org.sonarlint.intellij.finding.hotspot.NoBinding;
-import org.sonarlint.intellij.finding.hotspot.NoIssueSelected;
 import org.sonarlint.intellij.finding.hotspot.SecurityHotspotsStatus;
+import org.sonarlint.intellij.finding.hotspot.ValidStatus;
+import org.sonarlint.intellij.ui.nodes.LiveSecurityHotspotNode;
+import org.sonarlint.intellij.ui.tree.FlowsTree;
+import org.sonarlint.intellij.ui.tree.FlowsTreeModelBuilder;
+import org.sonarlint.intellij.ui.tree.SecurityHotspotTree;
+import org.sonarlint.intellij.ui.tree.SecurityHotspotTreeModelBuilder;
+import org.sonarlint.intellij.util.SonarLintActions;
 
 import static org.sonarlint.intellij.common.util.SonarLintUtils.getService;
+import static org.sonarlint.intellij.ui.SonarLintToolWindowFactory.createSplitter;
 
-public class SonarLintHotspotsListPanel {
-  private final JPanel mainPanel;
-  private final JBList<LiveSecurityHotspot> liveHotspotJBList;
-  private final Project project;
-  private final CardPanel cardPanel;
-  private LiveSecurityHotspot selectedHotspot;
-  private final SonarLintRulePanel sonarLintRulePanel;
+public class SonarLintHotspotsListPanel extends SimpleToolWindowPanel implements Disposable {
```

Comment:
I guess we should rename the class since it won't be a list anymore?

Reply:
Also we could get rid of the `SonarLint` prefix ?

Reply:
I removed SonarLint prefix